libweb3jsonrpc/Eth.cpp

```diff
@@ -271,6 +271,42 @@ string Eth::eth_sendTransaction(Json::Value const& _json)
 	{
 		throw;
 	}
+	catch (ZeroSignatureTransaction&)
+	{
+		BOOST_THROW_EXCEPTION(JsonRpcException("Zero signature transaction."));
+	}
+	catch (InvalidTransactionFormat&)
```

Comment:
Where can this be thrown from?

Reply:
I throw this in Client::submitTransaction in response to receiving ImportResult::ZeroSignature from TransactionQueue::import:
* Client::submitTransaction (from my PR branch): https://github.com/halfalicious/cpp-ethereum/blob/afe1842cf815b54c53ee3f3cfedfa523ebc60c51/libethereum/Client.cpp#L869-L875

* TransactionQueue::Import: 
https://github.com/ethereum/cpp-ethereum/blob/bcc98a31190e99fe536dc6ea97307fb1e4a188cb/libethereum/TransactionQueue.cpp#L79-L82

Reply:
But you throw `ZeroSignatureTransaction` there

Reply:
Ah sorry about that, I glanced at the code fragment you posted and saw the 

`BOOST_THROW_EXCEPTION(JsonRpcException("Zero signature transaction.")); `

so I thought you were referring to the `ZeroSignatureTransaction` exception rather than the `InvalidTransactionFormat` exception, my mistake!

Regarding `InvalidTransactionFormat`, it looks like this is only thrown in the `TransactionBase` constructor when a tx is being created from an RLP source: 

https://github.com/ethereum/cpp-ethereum/blob/bcc98a31190e99fe536dc6ea97307fb1e4a188cb/libethcore/TransactionBase.cpp#L47-L96

I don't think this code path can be hit by `Eth::eth_sendTransaction` (the version of `TransactionBase::TransactionBase` which accepts a `TransactionSkeleton` will be hit instead) so therefore I don't need to catch `InvalidTransactionFormat` at the RPC level. I'll remove the catch, thanks for pointing this out!